src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -566,6 +566,39 @@ private void OnSyncWithDefinitionEnd(NodeModel nodeModel)
             hasNodeInSyncWithDefinition = false;
         }
 
+        /// <summary>
+        /// Dispatches an action onto the UI thread. If there are no handlers for the underlying event
+        /// action is executed directly.
+        /// </summary>
+        /// <param name="a">the Action to execute</param>
+        /// <param name="blocking">If this action should run synchronously.</param>
+        internal void DispatchOnUIThread(Action a, bool blocking = false)
```

Comment:
If synchronous calls are the more common case, then should we just make the default value to be `true`?

Reply:
I thought about it. Probably. I was hoping it would encourage clients to ponder - can I make this asynchronous somehow, but maybe it will just cause hard to find bugs. I'll invert it.

Reply:
I would have been nice to have this logic in a lower level assembly....like DynamoUtilities